src/nat/tayga_helper.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ *    Copyright (c) 2017, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <openthread/border_router.h>
+
+#if OTBR_BORDER_ROUTING_NAT64
+#include "common/code_utils.hpp"
+#include "common/types.hpp"
+#include "nat/tayga_helper.hpp"
+#include "utils/system_utils.hpp"
+
+namespace otbr {
+namespace Tayga {
+
+static const char kTaygaConf[] = "/etc/tayga.conf";
+
+void ConfigTaygaNat64Prefix(otInstance *aInstance)
+{
+    otbrError   error = OTBR_ERROR_NONE;
+    otIp6Prefix nat64Prefix;
+    Ip6Prefix   prefix;
+
+    VerifyOrExit(otBorderRoutingGetNat64Prefix(aInstance, &nat64Prefix) == OT_ERROR_NONE,
+                 error = OTBR_ERROR_OPENTHREAD);
+
+    prefix.Set(nat64Prefix);
+    VerifyOrExit(SystemUtils::ExecuteCommand("sed -i '/^prefix /d' %s", kTaygaConf) == 0, error = OTBR_ERROR_ERRNO);
```

Comment:
This depends on commands may not be present on the system.

Reply:
Perhaps you can use [libconfig](http://www.hyperrealm.com/libconfig/libconfig_manual.html) instead.

Reply:
Thanks for point out. This nat64 prefix configuration will be enabled when border routing and nat64 are both supported, for which we expect to be ubuntu and raspbian as of now. And the sed command is expected to work just as in the _nat64 script.
libconfig usually works on structured conf files

Reply:
The scripts are supposed to run on ubuntu and rasbian, but the c++ binaries are not limited to these platforms.

Reply:
> Perhaps you can use [libconfig](http://www.hyperrealm.com/libconfig/libconfig_manual.html) instead.

Using `libconfig` seems too much dependency for just changing one line in a file?

Maybe we can generate the whole `tayga.conf` file from scratch in C++ code without using any system command or library?

Reply:
Updated to copy and rewrite the file.